--- Makefile~	2003-10-05 12:23:48.000000000 +0200
+++ Makefile	2004-05-19 01:12:53.360696848 +0200
@@ -127,13 +127,12 @@
 install: tcc_install libinstall
 
 tcc_install: tcc tcc.1 libtcc1.a bcheck.o
-	$(INSTALL) -m755 tcc $(bindir)
-	$(INSTALL) tcc.1 $(mandir)/man1
-	mkdir -p $(libdir)/tcc
-	mkdir -p $(libdir)/tcc/include
-	$(INSTALL) -m644 libtcc1.a bcheck.o $(libdir)/tcc
+	$(INSTALL) -d $(DESTDIR)$(bindir) $(DESTDIR)$(mandir)/man1 $(DESTDIR)$(libdir)/tcc/include
+	$(INSTALL) -m755 tcc $(DESTDIR)$(bindir)
+	$(INSTALL) tcc.1 $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m644 libtcc1.a bcheck.o $(DESTDIR)$(libdir)/tcc
 	$(INSTALL) -m644 stdarg.h stddef.h stdbool.h float.h varargs.h \
-                   tcclib.h $(libdir)/tcc/include
+                   tcclib.h $(DESTDIR)$(libdir)/tcc/include
 
 clean:
 	rm -f *~ *.o tcc tcc1 tcct tcc_g tcctest.ref *.bin *.i ex2 \
@@ -156,8 +156,9 @@
 
 # libtcc generation and example
 libinstall: libtcc.a 
-	$(INSTALL) -m644 libtcc.a $(libdir)
-	$(INSTALL) -m644 libtcc.h $(includedir)
+	$(INSTALL) -d $(DESTDIR)$(libdir) $(DESTDIR)$(includedir)
+	$(INSTALL) -m644 libtcc.a $(DESTDIR)$(libdir)
+	$(INSTALL) -m644 libtcc.h $(DESTDIR)$(includedir)
 
 libtcc.o: tcc.c i386-gen.c bcheck.c Makefile
 	$(CC) $(CFLAGS) -DLIBTCC -c -o $@ $<
